### PR TITLE
Development-MoM-batch-4

### LINKED ISSUE #468: 25506 - Implement Shadowed Acolyte v1.0
:memo: Implementation requested for Shadowed Acolyte (MoM)
---
_Date requested: Sun Jul 23 2023_

### What needs to be done?
New card needs to be added to this repository, which will require a card file to be created and potentially code implementation.
_Note: Non-text box & keyword implementations are automatically processed with updates to [throneteki-json-data]()._

Implementing Card (1.0)
:-------------------------:
image

---

### Implementation Steps
1. Create a branch from `development-MoM` (you may use a single branch for multiple implementations/updates)
2. Create & implement new card file for this card (within 'server/game/cards/MoM'). Existing cards can be used as templates
3. (Important) Set the card version (near bottom of card) to `ImplementedCardClass.version = '1.0'`
4. (Optional) Test the implementation locally
5. When ready (single or in bulk), create Pull Request into `development-MoM` branch, linking this issue in the description as 'Closes: #[this issue number]'

---

_:robot: Issue was automatically created, and will close itself when card implementation is pushed to the playtesting website._

### LINKED ISSUE #469: 25507 - Implement Deepwood Motte v1.0
## :memo: Implementation requested for Deepwood Motte (MoM)
_Date requested: Wed Aug 30 2023_

### What needs to be done?
New card needs to be added to this repository, which will require a card file to be created and potentially code implementation.
_Note: Non-text box & keyword implementations are automatically processed with updates to [throneteki-json-data]()._

Implementing Card (1.0)
:-------------------------:
image

---

### Implementation Steps
1. Create a branch from `development-MoM` (you may use a single branch for multiple implementations/updates)
2. Create & implement new card file for this card (within 'server/game/cards/MoM'). Existing cards can be used as templates
3. (Important) Set the card version (near bottom of card) to `ImplementedCardClass.version = '1.0'`
4. (Optional) Test the implementation locally
5. When ready (single or in bulk), create Pull Request into `development-MoM` branch, linking this issue in the description as 'Closes: #[this issue number]'

---

_:robot: Issue was automatically created, and will close itself when card implementation is pushed to the playtesting website._

### PR DESCRIPTION
**Pull request by LaranSoft**
Close #468 
Close #469

### CODE REVIEW

File: server/game/gamesteps/TriggeredAbilityWindow.js [hunk 1]
```diff
@@ -113,33 +113,58 @@ class TriggeredAbilityWindow extends BaseAbilityWindow {
...
+                        let card = this.getCardFromChoice(choice); return card && card.uuid === arg;
+                    });
+
+                    if(!choice) {
+                        return false;
+                    }
+
+                    this.chooseAbility(choice);
+                }
+
```
Comment (Patane97) [inline server/game/gamesteps/TriggeredAbilityWindow.js]: Not certain what this is doing; would remove unless absolutely vital (considering this is engine-level code).

File: server/game/gamesteps/TriggeredAbilityWindow.js [hunk 2]
```diff
@@ -113,33 +113,58 @@ class TriggeredAbilityWindow extends BaseAbilityWindow {
...
             return true;
         }
 
+        let unclickableCards = availableTargets.filter(card => card.location === 'active plot');
+
         this.game.promptForSelect(player, {
             activePromptTitle: `Choose triggering card for ${card.name}`,
             isCardEffect: false,
             cardCondition: card => availableTargets.includes(card),
+            additionalButtons: this.getButtons(player, unclickableCards),
```
Comment (Patane97) [inline server/game/gamesteps/TriggeredAbilityWindow.js]: I would not keep this here, as long as we specify the `cardType` below to allow for all clickable types; realistically, that's what was preventing us from clicking which plot to trigger from.

File: server/game/gamesteps/TriggeredAbilityWindow.js [hunk 3]
```diff
@@ -113,33 +113,58 @@ class TriggeredAbilityWindow extends BaseAbilityWindow {
...
 
-        let availableTargets = choices.map(choice => choice.context.event.card || choice.context.event.target).filter(card => !!card);
+        let availableTargets = choices.map(choice => this.getCardFromChoice(choice)).filter(card => !!card);
 
         if(choices.length === 1 || availableTargets.length <= 1) {
             this.chooseAbility(choices[0]);
             return true;
         }
 
+        let unclickableCards = availableTargets.filter(card => card.location === 'active plot');
```
Comment (Patane97) [inline server/game/gamesteps/TriggeredAbilityWindow.js]: Removing this as the below line it's used in should also be removed. :)

**Review by Patane97 — COMMENTED** [review]
Just made a few comments to help with learning - for the essence of time I've applied the adjustments myself, so you are welcome to check these comments & see the changes if it helps! :) Also can discuss in discord.

File: server/game/cards/25.7-MoM/ShadowedAcolyte.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const DrawCard = require('../../drawcard.js');
+
+class ShadowedAcolyte extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => this.game.getOpponents(this.controller).some(opponent => opponent.shadows.length > 0),
+            match: card => card === this,
```
[the remark this answers is not in the document]
Comment (LaranSoft) [reply server/game/cards/25.7-MoM/ShadowedAcolyte.js]: Much better! t y :)

File: server/game/cards/25.7-MoM/DeepwoodMotte.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
+const DrawCard = require('../../drawcard.js');
+const TextHelper = require('../../TextHelper');
+
+class DeepwoodMotte extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onPlotRevealed: event => event.plot.hasTrait('Winter') && !this.kneeled
```
[the remark this answers is not in the document]
Comment (LaranSoft) [reply server/game/cards/25.7-MoM/DeepwoodMotte.js]: Perfect 👍

**Pull request merged**